pkg/agent/cniserver/interface_configuration_linux.go

```diff
@@ -66,18 +69,54 @@ var (
 )
 
 type ifConfigurator struct {
+	hostProcPathPrefix          string
 	ovsDatapathType             ovsconfig.OVSDatapathType
 	isOvsHardwareOffloadEnabled bool
 	disableTXChecksumOffload    bool
 	netlink                     netlinkutil.Interface
 	sriovnet                    SriovNet
+	nsIDNameMapping             map[int]string
 }
 
-func newInterfaceConfigurator(ovsDatapathType ovsconfig.OVSDatapathType, isOvsHardwareOffloadEnabled bool, disableTXChecksumOffload bool) (*ifConfigurator, error) {
-	configurator := &ifConfigurator{ovsDatapathType: ovsDatapathType, isOvsHardwareOffloadEnabled: isOvsHardwareOffloadEnabled, disableTXChecksumOffload: disableTXChecksumOffload, netlink: &netlink.Handle{}, sriovnet: newSriovNet()}
+func newInterfaceConfigurator(ovsDatapathType ovsconfig.OVSDatapathType, isOvsHardwareOffloadEnabled bool, disableTXChecksumOffload bool, hostProcPathPrefix string) (*ifConfigurator, error) {
+	configurator := &ifConfigurator{ovsDatapathType: ovsDatapathType, isOvsHardwareOffloadEnabled: isOvsHardwareOffloadEnabled, disableTXChecksumOffload: disableTXChecksumOffload, netlink: &netlink.Handle{}, sriovnet: newSriovNet(), hostProcPathPrefix: hostProcPathPrefix}
+	configurator.getNsIDNameMapping()
 	return configurator, nil
 }
 
+const containerdNetNSFolder = "/var/run/netns"
+const dockerNetNSFolder = "/var/run/docker/netns"
```

Comment:
This may be not reliable and there are other container runtimes. It may be also complicated to mount so many host paths.
Could we save the netns path passed with CNI Add call and use it to enter the netns?

Reply:
Thanks for the quick review! Saving the `netns` passed via CNI requests should be a better approach. Do you have any suggestions on how to get the netns of the existing Pods created by previous versions?

Reply:
I feel this is a new enhancement which should be fine just supporting newly created pods. We could log INFO when the mtu updater doesn't find the desired netns.

Reply:
Got it. thanks! I will update this PR accordingly.